### PR TITLE
Connect Poster Screen to API to fetch random Posters

### LINKED ISSUE #32: Blurred Poster Screen
### Description

Once the player starts the game, the player is redirected to the Blurred Poster Screen. Update the current implementation and make it use data from the backend.

### Tasks

- [x] retrieve poster dynamically from api with graphql query
- [x] make use of animation library for unblurring the poster (10 seconds) [react-spring](https://www.react-spring.io/) or [ionic animations](https://ionicframework.com/docs/utilities/animations)
- [x] redirect the player to the next screen ([Question Screen](https://github.com/kwizapp/kwiz/issues/11)) if the screen gets tapped
- [x] record the time it took the user until the screen was tapped

### Context 
[Story: Blurred Poster Screen](https://github.com/kwizapp/kwiz/issues/9)

### PR DESCRIPTION
**Pull request by nikzaugg**
### Description

This PR adds the following:
* apollo client to create graphql queries
* `Poster.tsx` will display a random movie each time it's loaded. It gets the movie from our API
* new blur animation component
* clean up

### Related Issues

Issue: #57
[Story: Blurred Poster Screen](https://github.com/kwizapp/kwiz/issues/9)

### Checklist

- [x] Have you added tests where necessary? Do all the test pass? 
- [x] Have you added descriptive comments to your code?
- [x] Have you updated the documentation related to this propo

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -1,3 +1,6 @@
 module.exports = {
   extends: ['@kwizapp/eslint-config-ts-react'],
+  rules: {
+    '@typescript-eslint/member-delimiter-style': 'off',
```
Comment (rschlaefli) [inline .eslintrc.js]: Remove this, I had added it temporarily.

```suggestion

```

File: package.json [hunk 2]
```diff
@@ -33,12 +33,12 @@
...
     "eslint-config-prettier": "6.10.0",
     "eslint-plugin-import": "2.20.1",
-    "eslint-plugin-prettier": "^3.1.2",
+    "eslint-plugin-prettier": "3.1.2",
     "eslint-plugin-react": "7.19.0",
     "ionicons": "5.0.1",
     "prettier": "1.19.1",
-    "react": "16.13.0",
-    "react-dom": "16.13.0",
+    "react": "16.13.1",
```
Comment (rschlaefli) [inline package.json:66]: Not sure we should start using deps and devDeps. Wouldn't react be a dep as well? I would move everything to `dependencies`.

File: src/components/animations/BlurAnimated.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import { CreateAnimation } from '@ionic/react'
+import React, { useEffect, useRef } from 'react'
+
+interface Props {
+  duration?: number
+  children: JSX.Element
+}
+
+const BlurAnimated = ({ duration, children }: Props) => {
```
Comment (rschlaefli) [inline src/components/animations/BlurAnimated.tsx:9]: That is much better than my hacky class-based animation :D

File: src/components/loading/Loading.tsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        background:
+          'linear-gradient(0deg, rgba(173,220,244,1) 0%, rgba(235,250,248,1) 100%)',
+      }}
+    >
+      <IonGrid>
+        <IonRow>
+          <IonGrid>
+            <IonRow>
+              <IonCol>
+                <KwizAnimated />
```
Comment (rschlaefli) [inline src/components/loading/Loading.tsx:23]: What is this for? There is no children here that would be animated? :thinking:

File: src/pages/Poster.tsx [hunk 5]
```diff
@@ -1,81 +1,166 @@
...
-    setSecondsRemaining(QUESTION_TIME)
-    countDownRef.current = setInterval(() => {
-      setSecondsRemaining(prevSeconds => {
-        if (prevSeconds <= 1) {
-          return 0
-        }
-        return prevSeconds - 1
-      })
-    }, 1000)
+    console.log('useIonViewWillEnter', 'COUNTDOWN')
```
Comment (rschlaefli) [inline src/pages/Poster.tsx]: Remove the logs (I added for debugging)

**Review by rschlaefli — CHANGES_REQUESTED** [review]

File: src/components/loading/Loading.tsx (hunk 4 above)
Comment (nikzaugg) [reply src/components/loading/Loading.tsx:23]: What do yo mean? The `KwizAnimated` component is not a wrapper that you can out around other components. It's a component which shows the kwiz-logo and animates it using the keyframes that I have specified.

File: .eslintrc.js (hunk 1 above)
Comment (nikzaugg) [reply .eslintrc.js]: I've added this change in a separate Pull Request.
That one is merged now, so I can adjust here accordingly.

File: src/components/loading/Loading.tsx (hunk 4 above)
Comment (rschlaefli) [reply src/components/loading/Loading.tsx:23]: Aah. I was confused by the diff just above this and thought it was KwizAnimated. Instead it is `BlurAnimated`, which has a children prop.

File: package.json (hunk 2 above)
Comment (nikzaugg) [reply package.json:66]: done

File: src/pages/Poster.tsx (hunk 5 above)
Comment (nikzaugg) [reply src/pages/Poster.tsx]: done

**Review by rschlaefli — APPROVED** [review]
LGTM

**Pull request merged**